### PR TITLE
Adding getFile implementation

### PR DESCRIPTION
**Pull request by nkatzman**
This PR adds in the implementation of the `getFile` function

It takes in an scmUrl, token, path and a reference and will return the contents of the file that is found on github.

We probably needs to come up with an error translation function

### CODE REVIEW

**Comment by nkatzman** [comment]
Dependent on https://github.com/screwdriver-cd/scm-base/pull/3

File: index.js [hunk 1]
```diff
@@ -183,6 +183,47 @@ class GithubScm extends Scm {
...
+            token: config.token
+        });
+
+        return new Promise((resolve, reject) => {
+            this.breaker.runCommand({
+                action: 'getContent',
+                params: {
+                    user: scmInfo.user,
+                    repo: scmInfo.repo,
+                    path: config.path
```
Comment (d2lam) [inline index.js]: can you also add `branch` here for PR builds?

File: README.md [hunk 2]
```diff
@@ -64,6 +64,22 @@ The `updateCommitStatus` function will update the commit status for a given repo
...
 
+### getFile
+The parameters required are:
+
+| Parameter        | Type  |  Description |
+| :-------------   | :---- | :-------------|
+| config        | Object | Configuration Object |
+| config.scmUrl | String | The scmUrl to get permissions on |
+| config.token | String | The github token to check permissions on |
+| config.path | String | The path to the file on github to read |
```
Comment (nkatzman) [inline README.md]: add optional `ref` parameter to the documentation

Comment (shruthi-venkateswaran) [inline README.md]: Done

File: README.md [hunk 3]
```diff
@@ -64,6 +64,23 @@ The `updateCommitStatus` function will update the commit status for a given repo
...
+| config.ref | String | The reference to the github repo, could be a branch or sha |
+
+The `getFile` function will return the contents of a file read from github.
```
Comment (FenrirUnbound) [inline README.md]: I don't think you need this line. It's covered in the next line by how the promise resolves with the contents

File: test/index.test.js [hunk 4]
```diff
@@ -303,4 +304,151 @@ describe('index', () => {
             });
         });
     });
+
+    describe('getFile', () => {
+        const scmUrl = 'git@github.com:screwdriver-cd/models.git';
+        const content = `IyB3b3JrZmxvdzoKIyAgICAgLSBwdWJsaXNoCgpqb2JzOgogICAgbWFpbjoK\n
```
Comment (FenrirUnbound) [inline test/index.test.js]: Why did you use ticks for the string?

Comment (nkatzman) [inline test/index.test.js]: To make it a multi line string

Comment (FenrirUnbound) [inline test/index.test.js]: Ah gotcha 👍

**Comment by FenrirUnbound** [comment]
👍 

importantdownrighthoneybee

**Comment by tkyi** [comment]
👍 

giphy-1

**Pull request merged**